app/src/main/java/com/rkm/tasky/ui/component/ReminderDropDownMenu.kt

```diff
@@ -0,0 +1,163 @@
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReminderDropDownMenu(
+    state: ReminderDropDownMenuUiState,
+    actions: ReminderDropDownMenuActions,
+    modifier: Modifier = Modifier
+) {
+
+
+    ExposedDropdownMenuBox(
+        modifier = modifier,
+        expanded = state.isExpanded,
+        onExpandedChange = { actions.onExpanded() }
+    ) {
+        TextField(
+            modifier = Modifier.menuAnchor(MenuAnchorType.PrimaryNotEditable)
+                .fillMaxWidth(),
+            value = stringResource(state.selectedDuration.toUiString()),
+            onValueChange = {},
+            readOnly = true,
+            colors = ExposedDropdownMenuDefaults.outlinedTextFieldColors(
+                focusedBorderColor = ItemDetailReminderDividerColor,
+                unfocusedBorderColor = ItemDetailReminderDividerColor,
+            ),
+            leadingIcon = {
+                IconWithBackground(
+                    icon = {
+                        Icon(
+                            painter = painterResource(R.drawable.notications_icon),
+                            contentDescription = null,
+                            tint = ItemIconNotificationBellColor
+                        )
+                    },
+                    size = 30
+                )
+            }
+        )
+
+        ExposedDropdownMenu(
+            expanded = state.isExpanded,
+            onDismissRequest = { actions.onExpanded() },
+            containerColor = ItemDetailContainerColor
+        ) {
+            ReminderBeforeDuration.entries.forEach { option ->
+                DropdownMenuItem(
+                    text = {
+                        Text(
+                            text = stringResource(option.toUiString()),
+                            style = MaterialTheme.typography.bodyLarge
+                        )
+                    },
+                    onClick = {
+                        actions.onExpanded()
+                        actions.onSelectedDuration(option)
+                    },
+                    contentPadding = ExposedDropdownMenuDefaults.ItemContentPadding
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun IconWithBackground(
+    modifier: Modifier = Modifier,
+    icon: @Composable () -> Unit,
+    backgroundColor: Color = ItemIconNotificationBackgroundColor,
+    size: Int = 50
+) {
+
+    Box(
+        modifier = Modifier
+            .size(size.dp)
+            .clip(RoundedCornerShape(8.dp))
+            .background(backgroundColor),
+        contentAlignment = Alignment.Center
+    ) {
+        icon()
+    }
+
+}
+
+@Stable
+data class ReminderDropDownMenuUiState(
+    val isExpanded: Boolean = false,
+    val selectedDuration: ReminderBeforeDuration = ReminderBeforeDuration.TEN_MINUTES
+)
+
+@Stable
+data class ReminderDropDownMenuActions(
+    val onExpanded: () -> Unit,
+    val onSelectedDuration: (ReminderBeforeDuration) -> Unit
+)
```

Comment:
The Stable annotations here don't do anything, since primitive types and lambdas are already stable by default